packages/component/src/hooks/useRenderEmoji.js

```diff
@@ -0,0 +1,23 @@
+import useEmojiFromStyles from './useEmojiFromStyles';
+import { useTextBoxValue } from '../SendBox/TextBox';
+import { useCallback } from 'react';
+
+export default function useRenderEmoji() {
+  const [{ emojiAutocorrect, emojiUnicodeMap, emojiRegExp }] = useEmojiFromStyles();
+
+  const [value] = useTextBoxValue();
+
+  const setEmoji = useCallback(
+    value => {
+      const emoticon = value.match(emojiRegExp);
+
+      if (emojiAutocorrect && !!emoticon && !!emojiUnicodeMap[emoticon]) {
+        value = value.replace(emoticon, emojiUnicodeMap[emoticon]);
+      }
+      return value;
+    },
+    [emojiAutocorrect, emojiUnicodeMap, emojiRegExp]
+  );
+
+  return [value, setEmoji];
+}
```

Comment:
Please add a hook-specific test for this.

You can search `useXxx` for the set up for the hook-specific tests.

Also, update `HOOKS.md`.